Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -396,8 +396,21 @@ private void OnStateChanged(object requestor, CancellationToken token, ParserSta
             var handler = StateChanged;
             if (handler != null && !token.IsCancellationRequested)
             {
-                var args = new ParserStateEventArgs(state, oldStatus, token);
-                handler.Invoke(requestor, args);
+                try
+                {
+                    var args = new ParserStateEventArgs(state, oldStatus, token);
+                    handler.Invoke(requestor, args);
+                }
+                catch (Exception e)
+                {
+                    if (e is OperationCanceledException cancellation)
```

Comment:
Why do we use an if statement based on the type of exception instead of a separate `catch`?